Handle invalid length hex colors

Bobbar opened the pull request:
I'm working on a project to import data from xlsx sheets which are generated from an ERP system.  But the outputs from the ERP system contain color values which are not handled by the ColorStringParser. Specifically, there are many cells with the color value of "#0", which represents no color or transparent.  Excel opens these files without issue.

jahav commented:
@Bobbar I need a sample of the file, just piece/whole of styles.xml where the color is specified should be enough.

When I tried to specify `#0` as a color in the styles.xml, Excel always interpreted it as a black, not automatic color. Same happened for any non-standard  string.

Bobbar commented:
@jahav 

Here is a sample with the problematic color values.

[sample.xlsx](https://github.com/ClosedXML/ClosedXML/files/12917033/sample.xlsx)

It appears related to the font style specifically.   This is the xml style it seems to have trouble with: `<x:color rgb="0" tint="0" xmlns:x="http://schemas.openxmlformats.org/spreadsheetml/2006/main" />`

Thanks for looking at this!

Bobbar commented:
And yes, I believe you are correct about '0' representing black, not auto/transparent.

.FromArgb(0) would give the same value as .Empty, so that was my reasoning there. Apparently Excel interprets it differently.

jahav committed:
Modify color parser to handle invalid colors.

Modify color parser for ST_UnsignedIntHex so it doesn't throw on invalid colors and parses colors same way as Excel.

jahav committed:
Remove duplicated logic for parsing hex digit in color parser.
